extra-requirements.txt

```diff
@@ -55,6 +55,7 @@ cookiecutter:               hub, devel, cicd
 pytest:                     test
 pytest-xdist==1.34.0:       test
 pytest-timeout:             test
+pytest-mock:                test
```

Comment:
Since `unittest` is natively installed with python, I'd rather do the 
```
from unittest.mock import Mock
...
Mock(wrap=...)
```
instead of using pytest-mock. Apart from that, I like the PR.

Reply:
I chose the `pytest` version just because we are using all the `pytest` ecosystem, but no strong opinion about this.

@bwanglzu any consideration about this?

Reply:
`pytest-mock` is a good choice as we simply inject `mocker` in the argument and inside the test body we can explicitly define return values rather than having to patch on top of the function call (if we use Mock) and explicitly injecting the mocked patches - increases the number of arguments making the test look clumsy.